File: clients/fireback-react/src/sdk/fireback/modules/workspaces/AppMenuEntity.ts
import {
    BaseDto,
    BaseEntity,
} from "../../core/definitions"
import {
    CapabilityEntity,
} from "./CapabilityEntity"
// In this section we have sub entities related to this object
// Class body
export type AppMenuEntityKeys =
  keyof typeof AppMenuEntity.Fields;
export class AppMenuEntity extends BaseEntity {
  public href?: string | null;
  public icon?: string | null;
  public label?: string | null;
  public activeMatcher?: string | null;
  public applyType?: string | null;
  public capability?: CapabilityEntity | null;
      capabilityId?: string | null;
  public static Navigation = {
      edit(uniqueId: string, locale?: string) {
          return `${locale ? '/' + locale : ''}/app-menu/edit/${uniqueId}`;
      },
      create(locale?: string) {
          return `${locale ? '/' + locale : ''}/app-menu/new`;
      },
      single(uniqueId: string, locale?: string) {
          return `${locale ? '/' + locale : ''}/app-menu/${uniqueId}`;
      },
      query(params: any = {}, locale?: string) {
          return `${locale ? '/' + locale : ''}/app-menus`;
      },
      /**
      * Use R series while building router in CRA or nextjs, or react navigation for react Native
      * Might be useful in Angular as well.
      **/
      Redit: "app-menu/edit/:uniqueId",
      Rcreate: "app-menu/new",
      Rsingle: "app-menu/:uniqueId",
      Rquery: "app-menus",
  };
  public static definition = {
  "name": "appMenu",
  "http": {},
  "gormMap": {},
  "importList": [
    "modules/workspaces/CapabilityDefinitions.dyno.proto"
  ],
  "fields": [
    {
      "name": "href",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "icon",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "label",
      "type": "string",
      "translate": true,
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "activeMatcher",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "applyType",
      "type": "string",
      "computedType": "string",
      "gormMap": {}
    },
    {
      "name": "capability",
      "type": "one",
      "target": "CapabilityEntity",
      "computedType": "CapabilityEntity",
      "gormMap": {}
    }
  ],
  "cliDescription": "Manages the menus in the app, (for example tab views, sidebar items, etc.)",
  "cte": true
}
public static Fields = {
  ...BaseEntity.Fields,
      href: 'href',
      icon: 'icon',
      label: 'label',
      activeMatcher: 'activeMatcher',
      applyType: 'applyType',
          capabilityId: 'capabilityId',
      capability$: 'capability',
      capability: CapabilityEntity.Fields,
}
}

File: modules/workspaces/codegen/ios/firebackios/sdk/modules/workspaces/AppMenuEntity.swift
import Foundation
class AppMenuEntity : Codable, Identifiable {
    var href: String? = nil
    var icon: String? = nil
    var label: String? = nil
    var activeMatcher: String? = nil
    var applyType: String? = nil
    var capability: CapabilityEntity? = nil
    // var capabilityId: String? = nil
}
class AppMenuEntityViewModel: ObservableObject {
  // improve the fields here
  @Published var href: String? = nil
  @Published var hrefErrorMessage: String? = nil
  @Published var icon: String? = nil
  @Published var iconErrorMessage: String? = nil
  @Published var label: String? = nil
  @Published var labelErrorMessage: String? = nil
  @Published var activeMatcher: String? = nil
  @Published var activeMatcherErrorMessage: String? = nil
  @Published var applyType: String? = nil
  @Published var applyTypeErrorMessage: String? = nil
  func getDto() -> AppMenuEntity {
      var dto = AppMenuEntity()
    dto.href = self.href
    dto.icon = self.icon
    dto.label = self.label
    dto.activeMatcher = self.activeMatcher
    dto.applyType = self.applyType
      return dto
  }
}